### PR TITLE
no midair fire

### PR DESCRIPTION
**Pull request by akilan98**
Blocks dont get set on fire unless they are leaves or they have a block under them

### CODE REVIEW

**Comment by mastercoms** [comment]
why was glowkit changed?

**Comment by akilan98** [comment]
You committed over it while I was working. It shouldn't change.

**Comment by mastercoms** [comment]
Closing as fix is not needed.

File: src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java [hunk 1]
```diff
@@ -48,6 +48,10 @@ private boolean setBlockOnFire(GlowPlayer player, GlowBlock clicked, BlockFace f
             return true;
         }
 
+        if (clicked.getType() != Material.LEAVES && fireBlock.getRelative(BlockFace.DOWN).getType() == Material.AIR) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: Shouldn't this check if the type is not flammable? There are more blocks than leaves that allow for this.

**Comment by akilan98** [comment]
Need to add other side burnable things. (Flowers don't have an enum constant :( )

**Comment by akilan98** [comment]
Flowers don't get lit on the side

**Comment by kamcio96** [comment]
https://github.com/Bukkit/Bukkit/blob/master/src/main/java/org/bukkit/Material.java#L851 ?

**Comment by akilan98** [comment]
see this is invalid

**Comment by akilan98** [comment]
isFlammable checks if the block is flammable (dirt is) and then lights but it doesnt check if it's lighting on top of an object. It should not light in midair. It should however light on the side of some objects. If it's not an object that gets lit on the side there should be no fire

**Comment by mastercoms** [comment]
I don't see dirt in the switch statement for isFlammable..

**Comment by akilan98** [comment]
Actually the list I have should be replaced with isBurnable(). Those items can be lit on the side/bottom.

**Comment by jimmikaelkael** [comment]
Why not to use a static collection for the material list instead of doing so much OR ?

**Comment by akilan98** [comment]
@jimmikaelkael I want to use isBurnable() but it's not working when I tested...

**Pull request closed without merging**